graylog2-web-interface/src/components/common/Spinner.jsx

```diff
@@ -1,22 +1,52 @@
-import React from 'react';
+// @flow
+import React, { useState, useEffect } from 'react';
+import type { ComponentType } from 'react';
 import PropTypes from 'prop-types';
+import styled from 'styled-components';
 
 import Icon from './Icon';
 
+const Wrapper: ComponentType<{ visible: boolean }> = styled.span`
+  visibility: ${({ visible }) => (visible ? 'visible' : 'hidden')};
+`;
+
+type Props = {
+  delay?: number,
+  name?: string,
+  text?: string,
+}
+
 /**
  * Simple spinner to use while waiting for something to load.
  */
-const Spinner = ({ name, text, ...props }) => <span><Icon name={name} spin {...props} /> {text}</span>;
+const Spinner = ({ name, text, delay, ...rest }: Props) => {
+  const [delayFinished, setDelayFinished] = useState(false);
+
+  useEffect(() => {
+    const delayTimeout = window.setTimeout(() => {
+      setDelayFinished(true);
+    }, delay);
+
+    return () => clearTimeout(delayTimeout);
+  }, []);
+
+  return <Wrapper visible={delayFinished}><Icon name={name} spin {...rest} /> {text}</Wrapper>;
```

Comment:
maybe we just `return null` if the Spinner should not be visible so that we're not drawing invisible elements?